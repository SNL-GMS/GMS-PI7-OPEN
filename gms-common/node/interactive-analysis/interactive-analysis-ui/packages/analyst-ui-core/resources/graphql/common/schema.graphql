# Common GraphQL schema definitions shared across the components of the
# API Gateway (e.g. waveform, workflow)

# A date/time Type
scalar Date

# Represents a time range
# type TimeRange {
#     startTime: Float!
#     endTime: Float!
# }

# Input verion of a time range
input TimeRange {
    startTime: Float!
    endTime: Float!
}

# Represents a location specified using latitude (degrees), longitude (degrees),
# and altitude (kilometers).
type Location {

    # Latitude in degrees
    latDegrees: Float!
    
    # Longitude in degrees
    lonDegrees: Float!

    # Elevation in kilometers
    elevationKm: Float!
}

input LocationInput {

    # Latitude in degrees
    latDegrees: Float!
    
    # Longitude in degrees
    lonDegrees: Float!

    # Elevation in kilometers
    elevationKm: Float!
}

# Enumerated list of source types used to compute distances to
enum DistanceSourceType {
    Event
    UserDefined
}

# Represents a distance measurement relative to a specified source location
type DistanceToSource {

    # The distance in kilometers to the source
    distanceKm: Float!

    # The source location
    sourceLocation: Location!

    # The type of the source the distance is measured to (e.g. and event)
    sourceType: DistanceSourceType!

    # Optional: the unique ID of the source object
    sourceId: String!
}

input DistanceToSourceInput {
    # The source location
    sourceLocation: LocationInput

    # The type of the source the distance is measured to (e.g. and event)
    sourceType: DistanceSourceType

    # Optional: the unique ID of the source object
    sourceId: String
}

# Represents a frequency range
type FrequencyBand {
    # Minimum frequency in Herz
    minFrequencyHz: Float!

    # Maximum frequency in Herz
    maxFrequencyHz: Float!
}

# Represents a frequency range input to a mutation
input FrequencyBandInput {
    # Minimum frequency in Herz
    minFrequencyHz: Float!

    # Maximum frequency in Herz
    maxFrequencyHz: Float!
}

# Enumeration representing the different types of stations in the monitoring network
enum StationType {

    Seismic3Component,
    Seismic1Component,
    SeismicArray,
    Hydroacoustic,
    HydroacousticArray,
    Infrasound,
    InfrasoundArray,
    Weather,
    UNKNOWN
}

# Provenance reference details for information added to the System
type InformationSource {

    # Unique ID for the source
    id: String!

    # Source organization
    organizationName: String!

    # The time associated with the information source
    referenceTime: Float!
}

# Enumeration of configuration status changes reflected in the configuration
enum ConfigurationInfoStatus {

    # Initial configuration
    Initial

    # The system's installed location for the item associated with the configuration was corrected
    CorrectedSystemLocation

    # The physical location of the ite associated with the configuration was changed
    MovedPhysicalLocation

    # The item associated with the configuration was decommissioned
    Decommissioned
}

# Provenance information associated with a configuration item in the system
type ConfigurationInfo {

    # The unique ID of the user who created the Configuration
    creatorId: String!

    # The time the configuration took effect in the real world (e.g. station came online)
    effectiveTime: Float!

    # The time the configuration was installed (and thus took effect) in the System
    installationTime: Float!

    # The status change reflected in the configuration values
    status: ConfigurationInfoStatus!

    # Comment associated with the configuration information
    comment: String
}

# Enumerated type of the actor (e.g. analyst, system) associated with a CreationInfo object
enum CreatorType {
    Analyst
    System
}

# Provenance information about the results of data processing completed by the System
# and actions completed by users of the system
type CreationInfo {
    
    # The unique ID of the creation info object
    id: String!

    # The time (seconds since Unix epoch) the associated processing result was created
    creationTime: Float!

    # The unique ID of the user who created the associated processing result
    creatorId: String!

    # The enumerated type of the creator (e.g. analyst, system)
    creatorType: CreatorType!

    # TODO
    # The processing step associated with the processing result
    # processingStep: ProcessingStep!

    # TODO
    # The software component associated with the processing result
    # softwareComponent: SoftwareComponent!
}

# Represents an alias for a station, channel, site, etc.
type Alias {

    # The alias name
    name: String!

    # Provenance information about the alias
    configurationInfo: ConfigurationInfo
}

# Represents calibration information associated with a waveform
type ProcessingCalibration {

    # Calibration factor
    factor: Float!

    # Calibration factor error
    factorError: Float!

    # Calibration period
    period: Float!

    # Calibration time shift
    timeShift: Float!
}

# Input version of calibration information associated with a waveform
input ProcessingCalibrationInput {

    # Calibration factor
    factor: Float!

    # Calibration factor error
    factorError: Float!

    # Calibration period
    period: Float!

    # Calibration time shift
    timeShift: Float!
}

# Represents a generic timeseries interface containing the fields common to the
# concrete timeseries data types defined in the system (e.g. Waveform, FkPowerSpectrum, DetectionFeatureMap)
interface Timeseries {
    # Unique identifier for the timeseries
    id: String!

    # The start date/time (seconds since epoch) of the timeseries
    startTime: Float!

    # The end date/time (seconds since epoch) of the timeseries
    endTime: Float!

    # The sample rate (Hz) of the samples in the timeseries
    sampleRate: Float!

    # The sample count in the timeseries
    sampleCount: Int!
}

# Represents a time-bounded segment of waveform samples output from a channel (derived or raw).
# Waveforms implement the Timeseries interface
type Waveform implements Timeseries {

    # Unique identifier for the waveform
    id: String!

    # The start date/time (seconds since epoch) of the waveform
    startTime: Float!

    # The end date/time (seconds since epoch) of the waveform
    endTime: Float!

    # The sample rate (Hz) of the samples in the waveform
    sampleRate: Float!

    # The sample count in the waveform
    sampleCount: Int!

    # The time-ordered collection of waveform sample values
    waveformSamples: [Float]

    # Calibration information for the waveform
    calibration: ProcessingCalibration
}

# Input version of a time-bounded segment of waveform samples output from a channel (derived or raw).
# Waveforms implement the Timeseries interface
input WaveformInput {

    # Unique identifier for the waveform
    id: String!

    # The start date/time (seconds since epoch) of the waveform
    startTime: Float!

    # The end date/time (seconds since epoch) of the waveform
    endTime: Float!

    # The sample rate (Hz) of the samples in the waveform
    sampleRate: Float!

    # The sample count in the waveform
    sampleCount: Int!

    # The time-ordered collection of waveform sample values
    waveformSamples: [Float]

    # Calibration information for the waveform
    calibration: ProcessingCalibrationInput
}

# The following queries are supported
type Query {
    # Dummy
    queryCount: Int
}

# The following mutations are supported
type Mutation {
    # Dummy
    mutationCount: Int
}

# The following subscriptions are supported
type Subscription {
    # Dummy
    subscriptionCount: Int
}