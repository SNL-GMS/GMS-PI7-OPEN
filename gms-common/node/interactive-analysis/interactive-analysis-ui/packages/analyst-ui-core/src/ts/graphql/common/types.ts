// ***************************************
// Mutations
// ***************************************

// ***************************************
// Subscriptions
// ***************************************

// ***************************************
// Queries
// ***************************************

// ***************************************
// Model
// ***************************************

/**
 * Creation Type, reflects system change or analyst change
 */
export enum CreatorType {
  Analyst = 'Analyst',
  System = 'System'
}

/**
 * Distance value's units degrees or kilometers
 */
export enum DistanceUnits {
  degrees = 'degrees',
  km = 'km'
}

/**
 * Distance to source type
 */
export enum DistanceSourceType {
  Event = 'Event',
  UserDefined = 'UserDefined'
}

/**
 * Time range in epoch seconds
 */
export interface TimeRange {
  startTime: number;
  endTime: number;
}

/**
 * Location information
 */
export interface Location {
  latDegrees: number;
  lonDegrees: number;
  elevationKm: number;
}

/**
 * Position information relative to a location
 */
export interface Position {
  northDisplacementKm: number;
  eastDisplacementKm: number;
  verticalDisplacementKm: number;
}

/**
 * Log Level to determine different levels
 */
export enum LogLevel {
  INFO = 'INFO',
  DEBUG = 'DEBUG',
  WARNING = 'WARNING',
  ERROR = 'ERROR',
  DATA = 'DATA'
}

/**
 * Client Log Input
 */
export interface ClientLogInput {
  logLevel: LogLevel;
  message: string;
  time?: string;
}

/**
 * Client log mutation arguments
 */
export interface ClientLogMutationArgs {
  input: ClientLogInput;
}

/**
 * Creation Info
 */
export interface CreationInfo {
  id: string;
  creationTime: number;
  creatorId: string;
  creatorType: CreatorType;
  creatorName: string;
}

/**
 * Units used in DoubleValue part of feature prediction
 */
export enum Units {
    DEGREES = 'DEGREES',
    SECONDS= 'SECONDS',
    SECONDS_PER_DEGREE = 'SECONDS_PER_DEGREE',
    UNITLESS = 'UNITLESS'
}

/**
 * ENUM list of phase types
 */
export enum PhaseType {
  P = 'P',
  S = 'S',
  P3KPbc = 'P3KPbc',
  P4KPdf_B = 'P4KPdf_B',
  P7KPbc = 'P7KPbc',
  P7KPdf_D = 'P7KPdf_D',
  PKiKP = 'PKiKP',
  PKKSab = 'PKKSab',
  PKP2bc = 'PKP2bc',
  PKP3df_B = 'PKP3df_B',
  PKSab = 'PKSab',
  PP_1 = 'PP_1',
  pPKPbc = 'pPKPbc',
  PS = 'PS',
  Rg = 'Rg',
  SKiKP = 'SKiKP',
  SKKSac = 'SKKSac',
  SKPdf = 'SKPdf',
  SKSdf = 'SKSdf',
  sPdiff = 'sPdiff',
  SS = 'SS',
  sSKSdf = 'sSKSdf',
  Lg = 'Lg',
  P3KPbc_B = 'P3KPbc_B',
  P5KPbc = 'P5KPbc',
  P7KPbc_B = 'P7KPbc_B',
  Pb = 'Pb',
  PKKP = 'PKKP',
  PKKSbc = 'PKKSbc',
  PKP2df = 'PKP2df',
  PKPab = 'PKPab',
  PKSbc = 'PKSbc',
  PP_B = 'PP_B',
  pPKPdf = 'pPKPdf',
  PS_1 = 'PS_1',
  SKKP = 'SKKP',
  SKKSac_B = 'SKKSac_B',
  SKS = 'SKS',
  SKSSKS = 'SKSSKS',
  sPKiKP = 'sPKiKP',
  SS_1 = 'SS_1',
  SSS = 'SSS',
  nNL = 'nNL',
  P3KPdf = 'P3KPdf',
  P5KPbc_B = 'P5KPbc_B',
  P7KPbc_C = 'P7KPbc_C',
  PcP = 'PcP',
  PKKPab = 'PKKPab',
  PKKSdf = 'PKKSdf',
  PKP3 = 'PKP3',
  PKPbc = 'PKPbc',
  PKSdf = 'PKSdf',
  pPdiff = 'pPdiff',
  PPP = 'PPP',
  pSdiff = 'pSdiff',
  Sb = 'Sb',
  SKKPab = 'SKKPab',
  SKKSdf = 'SKKSdf',
  SKS2 = 'SKS2',
  Sn = 'Sn',
  sPKP = 'sPKP',
  SS_B = 'SS_B',
  SSS_B = 'SSS_B',
  NP = 'NP',
  P3KPdf_B = 'P3KPdf_B',
  P5KPdf = 'P5KPdf',
  P7KPdf = 'P7KPdf',
  PcS = 'PcS',
  PKKPbc = 'PKKPbc',
  PKP = 'PKP',
  PKP3ab = 'PKP3ab',
  PKPdf = 'PKPdf',
  Pn = 'Pn',
  pPKiKP = 'pPKiKP',
  PPP_B = 'PPP_B',
  pSKS = 'pSKS',
  ScP = 'ScP',
  SKKPbc = 'SKKPbc',
  SKP = 'SKP',
  SKS2ac = 'SKS2ac',
  SnSn = 'SnSn',
  sPKPab = 'sPKPab',
  sSdiff = 'sSdiff',
  NP_1 = 'NP_1',
  P4KPbc = 'P4KPbc',
  P5KPdf_B = 'P5KPdf_B',
  P7KPdf_B = 'P7KPdf_B',
  Pdiff = 'Pdiff',
  PKKPdf = 'PKKPdf',
  PKP2 = 'PKP2',
  PKP3bc = 'PKP3bc',
  PKPPKP = 'PKPPKP',
  PnPn = 'PnPn',
  pPKP = 'pPKP',
  PPS = 'PPS',
  pSKSac = 'pSKSac',
  ScS = 'ScS',
  SKKPdf = 'SKKPdf',
  SKPab = 'SKPab',
  SKS2df = 'SKS2df',
  SP = 'SP',
  sPKPbc = 'sPKPbc',
  sSKS = 'sSKS',
  P4KPdf = 'P4KPdf',
  P5KPdf_C = 'P5KPdf_C',
  P7KPdf_C = 'P7KPdf_C',
  Pg = 'Pg',
  PKKS = 'PKKS',
  PKP2ab = 'PKP2ab',
  PKP3df = 'PKP3df',
  PKS = 'PKS',
  PP = 'PP',
  pPKPab = 'pPKPab',
  PPS_B = 'PPS_B',
  pSKSdf = 'pSKSdf',
  Sdiff = 'Sdiff',
  SKKS = 'SKKS',
  SKPbc = 'SKPbc',
  SKSac = 'SKSac',
  SP_1 = 'SP_1',
  sPKPdf = 'sPKPdf',
  sSKSac = 'sSKSac',
  Sx = 'Sx',
  tx = 'tx',
  N = 'N',
  Px = 'Px',
  PKhKP = 'PKhKP',
  UNKNOWN = 'UNKNOWN'
}
