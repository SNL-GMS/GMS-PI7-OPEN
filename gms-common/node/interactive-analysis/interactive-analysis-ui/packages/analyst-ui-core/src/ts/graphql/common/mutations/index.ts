export { clientLog, clientLogMutation} from './client-log';
