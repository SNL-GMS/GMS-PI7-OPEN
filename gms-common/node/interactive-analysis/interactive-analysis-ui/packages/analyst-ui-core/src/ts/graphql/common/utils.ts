/**
 * Helper function to remove the __typename from object.
 * Gateway GraphQL doesn't like it.
 * @param object Generic object to strip __typename out of
 * @returns new object copied from param with __typename stripped out of
 */
export function removeTypeName(object: any): any {
    const newObj = {};
    // tslint:disable-next-line:forin
    for (const key in object) {
        if (object[key] instanceof Object) {
            newObj[key] = removeTypeName(object[key]);
        } else if (key !== '__typename') {
            newObj[key] = object[key];
        }
    }
    return newObj;
}
