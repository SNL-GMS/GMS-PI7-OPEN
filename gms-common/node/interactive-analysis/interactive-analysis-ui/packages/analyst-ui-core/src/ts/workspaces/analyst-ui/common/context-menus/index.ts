export { QcMaskContextMenu } from './qc-mask-context-menu';
export { SignalDetectionContextMenu } from './signal-detection-context-menu';
