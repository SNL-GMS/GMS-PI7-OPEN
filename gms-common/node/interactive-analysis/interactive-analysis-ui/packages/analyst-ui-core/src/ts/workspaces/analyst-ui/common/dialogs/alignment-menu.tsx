import { Radio, RadioGroup } from '@blueprintjs/core';
import { DropDown, FilterableOptionList } from '@gms/ui-core-components';
import * as React from 'react';
import { CommonTypes } from '~graphql/';
import { AlignWaveformsOn } from '../../components/waveform-display/types';

export interface AlignmentMenuProps {
  alignedOn: AlignWaveformsOn;
  phaseAlignedOn?: CommonTypes.PhaseType;
  sdPhases: CommonTypes.PhaseType[];
  prioritySdPhases?: CommonTypes.PhaseType[];

  onSubmit(alignedOn: AlignWaveformsOn, sdPhase?: CommonTypes.PhaseType);
}

export interface AlignmentMenuState {
  alignedOn: AlignWaveformsOn;
  phaseAlignedOn?: CommonTypes.PhaseType;
}

/**
 * A sub-set of the AlignWaveformsOn enum
 */
const alignmentsWithoutTime = {
  PREDICTED_PHASE: AlignWaveformsOn.PREDICTED_PHASE,
  OBSERVED_PHASE: AlignWaveformsOn.OBSERVED_PHASE
};

/**
 * How wide to render internal elements
 */
const widthPx = 160;

/**
 * The popup used to selected phases to align on
 */
export class AlignmentMenu extends React.Component<AlignmentMenuProps, AlignmentMenuState> {

  // If the drop down has already called onSubmit, do not call onsubmit in onBlur
  private hasBeenSubmitted: boolean = false;
  private constructor(props) {
    super(props);
    this.state = {
      alignedOn: this.props.alignedOn,
      phaseAlignedOn: this.props.phaseAlignedOn ? this.props.phaseAlignedOn : CommonTypes.PhaseType.P,
    };
  }

  /**
   * React component lifecycle.
   */
  public render() {
    const disablePhaseOptions = this.state.alignedOn === AlignWaveformsOn.TIME;

    return (
      <div
        className="alignment-dropdown"
      >
        <div className="alignment-dropdown__row">
          <RadioGroup
            onChange={this.handleRadioClick}
            selectedValue={
              this.state.alignedOn === AlignWaveformsOn.TIME ?
                'time' : 'phase'
            }
          >
            {
              this.state.alignedOn === AlignWaveformsOn.TIME ?
                <Radio
                  label="Time"
                  value="time"
                  key="time"
                  className="alignment-dropdown__radio"
                  autoFocus={true}
                  onClick={() => {this.submit(); }}
                />
                :
                <Radio
                  label="Time"
                  value="time"
                  key="time"
                  className="alignment-dropdown__radio"
                />
            }
            {
              <Radio
                label="Phase"
                value="phase"
                key="phase"
                className="alignment-dropdown__radio"
              />
            }
          </RadioGroup>
          <div className="alignment-dropdown__phase-type-selector">
            <DropDown
              value={
                this.state.alignedOn
              }
              dropDownItems={alignmentsWithoutTime}
              onMaybeValue={this.onPredictedOrObserved}
              disabled={disablePhaseOptions}
              widthPx={widthPx}
            />
          </div>

        </div>
        <div className="alignment-dropdown__list-positioner">
          <FilterableOptionList
            options={this.props.sdPhases}
            onSelection={this.onPhaseSelection}
            onDoubleClick={(phase: CommonTypes.PhaseType) => {this.submit(phase); }}
            onClick={(phase: CommonTypes.PhaseType) => {
              this.submit(phase);
            }}
            prioriotyOptions={this.props.prioritySdPhases}
            defaultSelection={this.props.phaseAlignedOn ? this.props.phaseAlignedOn : 'P'}
            widthPx={widthPx}
            disabled={disablePhaseOptions}
            onEnter={(phase: CommonTypes.PhaseType) => {this.submit(phase); }}
          />
        </div>
      </div>
    );
  }
  public componentWillUnmount() {
    if (!this.hasBeenSubmitted) {
      if (this.state.alignedOn !== this.props.alignedOn || this.state.phaseAlignedOn !== this.props.phaseAlignedOn) {
        this.submit(this.state.phaseAlignedOn);
      }
    }
  }

  /**
   * Returns current state of menu
   * 
   * @returns AlignmentMenuState
   */
  public getState = (): AlignmentMenuState => this.state.alignedOn === AlignWaveformsOn.TIME ?
    { alignedOn: this.state.alignedOn } :
    this.state

  private readonly onPredictedOrObserved = (val: AlignWaveformsOn) => {
    this.setState({ alignedOn: val });
  }

  private readonly handleRadioClick = (event: React.FormEvent<HTMLInputElement>) => {
    if (event.currentTarget.value === 'time') {
      this.hasBeenSubmitted = true;
      this.props.onSubmit(AlignWaveformsOn.TIME);
    } else {
      this.setState({ alignedOn: AlignWaveformsOn.PREDICTED_PHASE });
    }
  }

  /**
   * On phase selection handler.
   */
  private readonly onPhaseSelection = (phaseSelected: CommonTypes.PhaseType) => {
    this.setState({ phaseAlignedOn: phaseSelected });
  }

  private readonly submit = (phase?: CommonTypes.PhaseType) => {
    this.hasBeenSubmitted = true;
    this.props.onSubmit(this.state.alignedOn, phase);
  }
}
