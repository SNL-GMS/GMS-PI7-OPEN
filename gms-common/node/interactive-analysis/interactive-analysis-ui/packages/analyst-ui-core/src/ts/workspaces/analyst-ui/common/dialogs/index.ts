export { QcMaskFilter } from './qc-mask-filter';
export { QcMaskForm } from './qc-mask-form';
export { QcMaskOverlap } from './qc-mask-overlap';
export { SignalDetectionDetails } from './signal-detection-details';
export { AlignmentMenu } from './alignment-menu';
export { PhaseSelectionMenu } from './phase-selection-menu';
