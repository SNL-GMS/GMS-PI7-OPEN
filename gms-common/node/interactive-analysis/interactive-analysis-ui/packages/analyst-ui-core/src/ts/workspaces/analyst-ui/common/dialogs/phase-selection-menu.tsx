import { FilterableOptionList } from '@gms/ui-core-components';
import * as React from 'react';
import { CommonTypes } from '~graphql/';

/**
 * How wide to render internal elements
 */
const widthPx = 160;

export interface PhaseSelectionMenuProps {
  phase?: CommonTypes.PhaseType;
  sdPhases: CommonTypes.PhaseType[];
  prioritySdPhases?: CommonTypes.PhaseType[];

  onBlur(phase: CommonTypes.PhaseType);
  onEnterForPhases?(phase: CommonTypes.PhaseType);
  onPhaseClicked?(phase: CommonTypes.PhaseType);
}

export interface PhaseSelectionMenuState {
  phase: CommonTypes.PhaseType;
}

/**
 * Phase selection menu.
 */
export class PhaseSelectionMenu extends
  React.Component<PhaseSelectionMenuProps, PhaseSelectionMenuState> {

  private constructor(props) {
    super(props);
    this.state = {
      phase: this.props.phase ? this.props.phase : CommonTypes.PhaseType.P
    };
  }
  /**
   * React component lifecycle.
   */
  public render() {
    return (
      <div
        className="alignment-dropdown"
      >
        <FilterableOptionList
          options={this.props.sdPhases}
          onSelection={this.onPhaseSelection}
          onClick={this.onClick}
          onDoubleClick={this.onClick}
          prioriotyOptions={this.props.prioritySdPhases}
          defaultSelection={this.props.phase}
          widthPx={widthPx}
          onEnter={this.props.onEnterForPhases}
        />
      </div>
    );
  }

  /**
   * Returns current state of menu
   * 
   * @returns PhaseSelectionMenuState
   */
  public getState = (): PhaseSelectionMenuState =>
    this.state

  /**
   * On phase selection event handler.
   * 
   * @param phase the selected phase
   */
  private readonly onPhaseSelection = (phase: CommonTypes.PhaseType) => {
    this.setState({ phase });
  }

  private readonly onClick = (phase: CommonTypes.PhaseType) => {
    if (this.props.onPhaseClicked) {
        this.props.onPhaseClicked(phase);
    } else {
      this.setState({ phase });
    }
  }
}
