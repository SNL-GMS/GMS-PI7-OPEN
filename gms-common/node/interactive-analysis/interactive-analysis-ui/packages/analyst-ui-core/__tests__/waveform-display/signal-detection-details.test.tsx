import * as React from 'react';
import { SignalDetectionDetails } from '~analyst-ui/common/dialogs';
import { signalDetectionsData } from '../__data__/signal-detections-data';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

it('SignalDetectionDetails renders & matches snapshot', () => {
  const wrapper = Enzyme.shallow(
    <SignalDetectionDetails detection={signalDetectionsData[0]} color={'red'} />
  );
  expect(wrapper)
    .toMatchSnapshot();
});
