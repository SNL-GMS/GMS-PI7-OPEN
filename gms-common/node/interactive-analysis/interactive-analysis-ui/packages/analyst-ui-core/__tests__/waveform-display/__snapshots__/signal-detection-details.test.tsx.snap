// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignalDetectionDetails renders & matches snapshot 1`] = `
<div>
  <Form
    defaultPanel={
      Object {
        "formItems": Array [
          Object {
            "displayText": "P",
            "itemKey": "Phase",
            "itemType": "Display",
            "labelText": "Phase",
          },
          Object {
            "displayText": "2010/05/20T22:07:17.850000",
            "displayTextFormat": "Time",
            "itemKey": "Detection time",
            "itemType": "Display",
            "labelText": "Detection time",
          },
          Object {
            "displayText": "",
            "itemKey": "Time uncertainty",
            "itemType": "Display",
            "labelText": "Time uncertainty",
          },
          Object {
            "displayText": "creatorId",
            "itemKey": "Author",
            "itemType": "Display",
            "labelText": "Author",
          },
          Object {
            "displayText": "2018/12/05T21:25:50.378000",
            "displayTextFormat": "Time",
            "itemKey": "Creation time",
            "itemType": "Display",
            "labelText": "Creation time",
          },
          Object {
            "displayText": "No",
            "itemKey": "Rejected",
            "itemType": "Display",
            "labelText": "Rejected",
          },
        ],
        "key": "Current Version",
      }
    }
    disableSubmit={true}
    extraPanels={
      Array [
        Object {
          "content": <div
            className="ag-theme-dark signal-detection-details-versions-table"
          >
            <div
              className="max"
            >
              <Table
                columnDefs={
                  Array [
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "enableCellChangeFlash": true,
                      "field": "creationTime",
                      "headerName": "Creation time",
                      "valueFormatter": [Function],
                      "width": 165,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "enableCellChangeFlash": true,
                      "field": "phase",
                      "headerName": "Phase",
                      "width": 70,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "enableCellChangeFlash": true,
                      "field": "arrivalTimeMeasurementTimestamp",
                      "headerName": "Detection time",
                      "valueFormatter": [Function],
                      "width": 165,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "enableCellChangeFlash": true,
                      "field": "arrivalTimeMeasurementUncertaintySec",
                      "headerName": "Time uncertainty",
                      "valueFormatter": [Function],
                      "width": 125,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "field": "rejected",
                      "headerName": "Rejected",
                      "valueFormatter": [Function],
                      "width": 75,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "enableCellChangeFlash": true,
                      "field": "author",
                      "headerName": "Author",
                      "width": 90,
                    },
                  ]
                }
                getRowNodeId={[Function]}
                overlayNoRowsTemplate="No Verions"
                rowClassRules={
                  Object {
                    "versions-table__row--first-in-table": [Function],
                  }
                }
                rowData={
                  Array [
                    Object {
                      "arrivalTimeMeasurementFeatureType": "ARRIVAL_TIME",
                      "arrivalTimeMeasurementTimestamp": 1274393334.1,
                      "arrivalTimeMeasurementUncertaintySec": 0.685,
                      "author": "creatorId",
                      "creationTime": 1544048130.971,
                      "first-in-table": true,
                      "id": "b90ee9c8-3fc9-4ccd-95ea-5d1d206c35a7",
                      "phase": "tx",
                      "rejected": false,
                      "versionId": "b90ee9c8-3fc9-4ccd-95ea-5d1d206c35a7",
                    },
                    Object {
                      "arrivalTimeMeasurementFeatureType": "ARRIVAL_TIME",
                      "arrivalTimeMeasurementTimestamp": 1274393334.1,
                      "arrivalTimeMeasurementUncertaintySec": 0.685,
                      "author": "creatorId",
                      "creationTime": 1544048130.971,
                      "id": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
                      "phase": "P",
                      "rejected": false,
                      "versionId": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
                    },
                  ]
                }
                rowSelection="single"
              />
            </div>
          </div>,
          "key": "All Versions",
        },
      ]
    }
    header="Signal Detection"
    headerDecoration={
      <div
        className="signal-detection-swatch"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    }
    onCancel={[Function]}
  />
</div>
`;
